iree/hal/vulkan/dynamic_symbols.cc

```diff
@@ -132,22 +133,43 @@ StatusOr<ref_ptr<DynamicSymbols>> DynamicSymbols::Create(
 StatusOr<ref_ptr<DynamicSymbols>> DynamicSymbols::CreateFromSystemLoader() {
   IREE_TRACE_SCOPE0("DynamicSymbols::CreateFromSystemLoader");
 
-#if defined(IREE_VK_ICD_FILENAMES)
 #define IREE_STRINGIFY_(x) #x
 #define IREE_STRING_(x) IREE_STRINGIFY_(x)
+
+#if defined(IREE_VK_ICD_FILENAMES)
   std::string vk_icd_filenames = IREE_STRING_(IREE_VK_ICD_FILENAMES);
-#undef IREE_STRINGIFY_
-#undef IREE_STRING_
 #if defined(IREE_PLATFORM_WINDOWS)
-  // TODO(b/138220713): Set VK_ICD_FILENAMES on Windows
+// TODO(scotttodd): Set VK_ICD_FILENAMES on Windows
 #else
   ::setenv("VK_ICD_FILENAMES", vk_icd_filenames.c_str(), 0);
 #endif  // IREE_PLATFORM_WINDOWS
 #else
-  // Leave VK_ICD_FILENAMES unchanged and rely on the system Vulkan loader to
-  // discover ICDs.
+// Leave VK_ICD_FILENAMES unchanged and rely on the system Vulkan loader to
+// discover ICDs.
 #endif  // IREE_VK_ICD_FILENAMES
 
+#if defined(IREE_VK_LAYER_PATH_EXTRAS)
+  std::string vk_layer_path_extras = IREE_STRING_(IREE_VK_LAYER_PATH_EXTRAS);
+#if defined(IREE_PLATFORM_WINDOWS)
+// TODO(scotttodd): Layer discovery on Windows
+#else
+  const char* vk_layer_path = ::getenv("VK_LAYER_PATH");
+  std::string combined_path;
+  if (vk_layer_path) {
+    combined_path = absl::StrCat(vk_layer_path, ":", vk_layer_path_extras);
+  } else {
+    combined_path = vk_layer_path_extras;
+  }
+  ::setenv("VK_LAYER_PATH", combined_path.c_str(), 0);
```

Comment:
*shudder*

Reply:
Yeah >___> ...

The `VK_ICD_FILENAMES` code is used for SwiftShader (mostly Google upstream where it's harder to set environment variables outside of a program). This new code for `VK_LAYER_PATH`  is used across environments for `VK_LAYER_KHRONOS_timeline_semaphore`. It's gross and I'd rather inject the layer into our binaries somehow, without going through the loader.